File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.go
func intersection(nums1 []int, nums2 []int) []int {
	hashMap := make (map[int]bool)
	for _, num := range nums1 {
		hashMap[num] = false
	}
	for _, num := range nums2 {
		if _, ok := hashMap[num]; ok {
			hashMap[num] = true
		}
	}
	ans := make([]int, 0)
	for k, v := range hashMap {
		if v {
			ans = append(ans, k)
		}
	}
	return ans
}

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Map<Integer, Boolean> hashMap = new HashMap<>();
        for (int num: nums1) {
            hashMap.put(num, false);
        }
        for (int num: nums2) {
            hashMap.computeIfPresent(num, (k, v) -> true);
        }
        List<Integer> answerList = new ArrayList<>();
        hashMap.forEach((k,v) -> {
            if (v) {
                answerList.add(k);
            }
        });
        int[] ans = new int[answerList.size()];
        for (int i = 0; i < answerList.size(); i++) {
            ans[i] = answerList.get(i);
        }
        return ans;
    }
}